internal/resources/aws/data_transfer.go

```diff
@@ -0,0 +1,278 @@
+		Unit:            "GB",
+		UnitMultiplier:  decimal.NewFromInt(1),
+		MonthlyQuantity: decimalPtr(networkUsage),
+		ProductFilter:   r.buildProductFilter("AWS Outbound", nil),
+		PriceFilter: &schema.PriceFilter{
+			EndUsageAmount: strPtr(endUsageAmount),
+		},
+	}
+}
+
+// outboundUsEastCostComponents returns a cost component for outbound data
+// transfer to US East regions only when its usage is specified.
+func (r *DataTransfer) outboundUsEastCostComponents() []*schema.CostComponent {
+	costComponents := []*schema.CostComponent{}
+
+	if r.MonthlyOutboundUsEastToUsEastGB == nil {
+		return costComponents
+	}
+
+	toRegion := "us-east-1"
+
+	if r.Region == "us-east-1" {
+		toRegion = "us-east-2"
+	}
+
+	costComponents = append(costComponents, &schema.CostComponent{
+		Name:            "Outbound data transfer to US East regions",
+		Unit:            "GB",
+		UnitMultiplier:  decimal.NewFromInt(1),
+		MonthlyQuantity: decimalPtr(decimal.NewFromFloat(*r.MonthlyOutboundUsEastToUsEastGB)),
+		ProductFilter:   r.buildProductFilter("InterRegion Outbound", &toRegion),
+	})
+
+	return costComponents
+}
+
+// outboundOtherRegionsCostComponents returns a cost component for outbound data
+// transfer to other regions only when its usage is specified.
+func (r *DataTransfer) outboundOtherRegionsCostComponents() []*schema.CostComponent {
+	costComponents := []*schema.CostComponent{}
+
+	if r.MonthlyOutboundOtherRegionsGB == nil {
+		return costComponents
+	}
+
+	toRegion := "us-west-1"
+
+	switch r.Region {
+	case "us-east-1":
+		toRegion = "us-west-2"
+	case "us-west-1":
+		toRegion = "us-west-2"
+	case "cn-north-1":
+		toRegion = "cn-northwest-1"
+	case "cn-northwest-1":
+		toRegion = "cn-north-1"
+	}
+
+	costComponents = append(costComponents, &schema.CostComponent{
+		Name:            "Outbound data transfer to other regions",
+		Unit:            "GB",
+		UnitMultiplier:  decimal.NewFromInt(1),
+		MonthlyQuantity: decimalPtr(decimal.NewFromFloat(*r.MonthlyOutboundOtherRegionsGB)),
+		ProductFilter:   r.buildProductFilter("InterRegion Outbound", &toRegion),
+	})
+
+	return costComponents
+}
+
+// buildProductFilter returns a filter for data transfer products. Desctination
+// region is optional.
+func (r *DataTransfer) buildProductFilter(transferType string, toRegion *string) *schema.ProductFilter {
+	fromLocation := RegionMapping[r.Region]
+
+	attributeFilters := []*schema.AttributeFilter{
+		{Key: "transferType", Value: strPtr(transferType)},
+		{Key: "fromLocation", Value: strPtr(fromLocation)},
+	}
+
+	if toRegion != nil {
+		toLocation := RegionMapping[*toRegion]
+		attributeFilters = append(attributeFilters, &schema.AttributeFilter{
+			Key:   "toLocation",
+			Value: strPtr(toLocation),
+		})
+	}
+
+	if regionCode, ok := RegionCodeMapping[r.Region]; ok {
+		fmt.Println()
+		attributeFilters = append(attributeFilters, &schema.AttributeFilter{
+			Key:        "usagetype",
+			ValueRegex: strPtr(fmt.Sprintf("/^%s-/i", regionCode)),
+		})
+	}
```

Comment:
This is the fix. As `RegionCodeMapping` includes only one key for now, it won't affect other regions and it could be reused in future if needed.